tempest-testing-internal/src/main/kotlin/app/cash/tempest/testing/internal/TestUtils.kt

```diff
@@ -29,34 +29,64 @@ import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
 import com.amazonaws.services.dynamodbv2.document.DynamoDB
 import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput
 import java.net.ServerSocket
+import java.net.Socket
 
 fun pickRandomPort(): Int {
   ServerSocket(0).use { socket -> return socket.localPort }
 }
 
+fun serverListening(host: String, port: Int): Boolean {
+  var s: Socket? = null
+  return try {
+    s = Socket(host, port)
+    true
+  } catch (e: Exception) {
+    false
+  } finally {
+    if (s != null) try {
+      s.close()
+    } catch (e: Exception) {
+    }
```

Comment:
Feels like bottomline we should at least log the exception.

Reply:
In both places.